### DISCUSSION TITLE
One to many relationship

### DISCUSSION THREAD

**Discussion by NisthaAdhikari** [dynamoose/dynamoose#1299] [Q&A]
I have two tables i.e Table and Reservation. To create relation between them, I have added tableId attribute in Reservation which is of type Model type (of type Table type), as mentioned in the docs. Using the document.populate I am able to get the Table data through the tableId attribute from Reservation table. But how can I retrieve all Reservation from a Table? (Reservation and Table has one to many relation)?

These are my Model: 
Table Model:

```
const tableSchema = new Schema ({
        tableId: {
            type: String,
            required: true,
            unique: true,
            hashKey: true
        },
        name: {
             type: String,
             default: null
        },
    });
```

*Reservation Model:*

   
```
 const reservationSchema = new Schema ({
        id: {
            type: Number,
            required: true,
            unique: true,
            hashKey: true
        },
        tableId: table,   \\as per doc attribute of Table (Model) type 
        date: {
            type: String
        }
    });

```
This is how I retrieve table data from reservation model

```
reservationModel.scan().exec()
        .then(posts => {
            return posts.populate({
                path: 'tableId',
                model: 'Space'
              });
        })
        .then(populatedPosts => {
        
            console.log('pp',populatedPosts);
            return {
                allData: {
                  message: "Executedddd succesfully",
                  data: populatedPosts
                }
              }
        })
```

**Comment by fishcharlie** [comment]
I have converted this to a Discussion since it is not an Issue.

Below is the same answer I provided on your [Stack Overflow question](https://stackoverflow.com/a/70114645/894067) (however it **might** not contain all edits and comments made, and vice versa).

---

As of v2.8.2, Dynamoose does not support this. Dynamoose is focused on one directional simple relationships. This is partly due to the fact that we discourage use of `model.populate`. It is important to note that `model.populate` does another completely separate request to DynamoDB. This increases the latency and decreases the performance of your application.

DynamoDB truly requires a shift in how you think about modeling your data compared to SQL. I recommend watching [AWS re:Invent 2019: Data modeling with Amazon DynamoDB (CMY304)](https://youtu.be/DIQVJqiSUkE) for a great explanation of how you can model your data in DynamoDB in a highly efficient manner.

At some point Dynamoose might add support for this, but it's really hard to say if we will.

If you truly want to do this, I'd recommend adding a global index to your `tableId` property in your reservation schema. Then you can run something like the following:

```js
async function code(id) {
    const reservation = await reservationModel.get(id);
    const tables = await tableModel.query("tableId").eq(id).exec(); // This will be an array of `table` entries where `"tableId"=id`. Remember, it is required you add an index for this to work.
}
```

Remember, this will cause multiple calls to DynamoDB and isn't as efficient. I'd highly recommend watching the video I linked above to get more information about how to model your data in an more efficient manner.

Finally, I'd like to point out that your `unique: true` code does nothing. As seen in the [Dynamoose Attribute Settings Documentation](https://dynamoosejs.com/guide/Schema#attribute-settings), `unique` is not a valid setting. In your case since you don't have a `rangeKey`, it's not possible for two items to have the same `hashKey`, so technically it's already a unique property based on that. However it is important to note that you can overwrite existing items when creating an item. You can set `overwrite` to `false` for [`document.save`](https://dynamoosejs.com/guide/Document#documentsavesettings) or [`Model.create`](https://dynamoosejs.com/guide/Model#modelcreatedocument-settings-callback) to prevent that behavior and throw an error instead of overwriting your document.
